src/content/en/updates/2018/10/animation-worklet.md

```diff
@@ -0,0 +1,493 @@
+should keep that in mind when writing animation worklet code. Since all
+mathematical operations can handle `NaN` (they return `NaN` when one of their
+inputs is `NaN`) we are fine here!
+
+We are taking the `Math.sin` of the `currentTime`, and remapping that value to
+the range [0; 2000], which is the time range that our effect is defined for. Now
+[the animation looks very
+different](https://animation-worklet-article.glitch.me/sin.html), without having
+changed the keyframes or the animation's options. The worklet code can be
+arbitrarily complex, and allows you to programmatically define which effects are
+played in which order and to which extent.
+
+<video loop muted autoplay
+  src="https://storage.googleapis.com/webfundamentals-assets/animation-worklet/sin_x264.mp4">
+</video>
+
+## Options over Options
+
+You might find yourself wanting to reuse a worklet and just wanting to adjust
+some numbers. For this reason the WorkletAnimation constructor allows you pass
+an options object to the worklet:
+
+    registerAnimator('factor', class {
+      constructor(options = {}) {
+        this.factor = options.factor || 1;
+      }
+      animate(currentTime, effect) {
+        effect.localTime = currentTime * this.factor;
+      }
+    });
+
+    new WorkletAnimation(
+      'factor',
+      new KeyframeEffect(
+        document.querySelector('#b'),
+        [ /* ... same keyframes as before ... */ ],
+        {
+          duration: 2000,
+          iterations: Number.POSITIVE_INFINITY
+        }
+      ),
+      document.timeline,
+      {factor: 0.5}
+    ).play();
+
+Note: The options object will be structurally cloned when it is being sent to
+the worklet, similar to how `postMessage()` operates.
+
+In this [example](https://animation-worklet-article.glitch.me/options.html),
+both animations are driven with the same worklet, but with different options.
+
+## Gimme your local state!
+
+As I hinted at before, one of the key problems AnimationWorklet aims to solve is
+stateful animations. AnimationWorklets are allowed to hold state. However, one
+of the core features of worklets is that they can be migrated to a different
+thread or even be destroyed to save resources, which would also destroy the
+state. To prevent this kind of state loss, AnimationWorklet offers a hook that
+is called _before_ a worklet is destroyed that you can use to return a state
+object. That object will be passed to the constructor when the worklet is
+re-created. On initial creation, that parameter will be `undefined`.
+
+    registerAnimator('randomspin', class {
+      constructor(options = {}, state = {}) {
+        this.direction = state.direction || (Math.random() > 0.5 ? 1 : -1);
+      }
+      animate(currentTime, effect) {
+        // Some math to make sure that `localTime` is always > 0.
+        effect.localTime = 2000 + this.direction * (currentTime % 2000);
+      }
+      destroy() {
```

Comment:
(I wish we could get this in JS classes. It would solve a concurrency problem I'm having at the moment.)